README.md

```diff
@@ -261,27 +267,81 @@ function createConnectQueryKey<Desc extends DescMethod | DescService>(
 
 This function is used under the hood of `useQuery` and other hooks to compute a [`queryKey`](https://tanstack.com/query/v4/docs/react/guides/query-keys) for TanStack Query. You can use it to create (partial) keys yourself to filter queries.
 
-### `createConnectInfiniteQueryKey`
+`useQuery` creates a query key with the following parameters:
+1. The qualified name of the RPC.
+2. The transport being used.
+3. The request message.
+
+To create the same key manually, you simply provide the same parameters:
 
 ```ts
-function createConnectInfiniteQueryKey<
-  I extends Message<I>,
-  O extends Message<O>,
->(
-  methodDescriptor: Pick<MethodUnaryDescriptor<I, O>, "I" | "name" | "service">,
-  input: SkipToken | PartialMessage<I>,
-  pageParamKey: keyof PartialMessage<I>,
-): ConnectInfiniteQueryKey<I>;
+import { createConnectQueryKey, useTransport } from "@connectrpc/connect-query";
+import { ElizaService } from "./gen/eliza_pb";
+
+const myTransport = useTransport();
+const queryKey = createConnectQueryKey({
+  // The schema is the only required parameter. 
+  schema: ElizaService.method.say,
+  transport: myTransport,
+  // You can provide a partial message here.
+  input: { sentence: "hello" },
+});
+
+// queryKey:
+[
+  "conect-query",
+  {
+    transport: "t1",
+    serviceName: "connectrpc.eliza.v1.ElizaService",
+    methodName: "Say",
+    input: { sentence: "hello" },
+    cardinality: "finite",
+  }
+]
 ```
 
-This function is not really necessary unless you are manually creating infinite query keys. When invalidating queries, it usually makes more sense to use the `createConnectQueryKey` function instead since it will also invalidate the regular queries (as well as the infinite queries).
+You can create a partial key that matches all RPCs of a service:
+
+```ts
+import { createConnectQueryKey } from "@connectrpc/connect-query";
+import { ElizaService } from "./gen/eliza_pb";
+
+const queryKey = createConnectQueryKey({
+  schema: ElizaService,
+});
+
+// queryKey:
+[
+  "conect-query",
+  {
+    serviceName: "connectrpc.eliza.v1.ElizaService",
+    cardinality: "finite",
+  }
+]
+```
+
+Infinite queries have distinct keys. To create a key for an infinite query, use the parameter `cardinality`:
+
+```ts
+import { createConnectQueryKey } from "@connectrpc/connect-query";
+import { ListService } from "./gen/list_pb";
+
+// The hook useInfiniteQuery() creates a query key with cardinality: "infinite",
+// and passes on the pageParamKey.
+const queryKey = createConnectQueryKey({
+  schema: ListService.method.list,
+  cardinality: "infinite", // "any" matches infinite and finite queries
```

Comment:
Not sure what this means by "any". I think if we leave off cardinality, it'll match both. But that would be undefined no?

Reply:
Ah I see. Yes I was expecting cardinality for the keys to not default to finite. I actually think default to undefined cardinality is more consistent with existing behaviour since the default was to ignore the final "infinite" section anyways. I can make this change.

Reply:
Hmm, unless you were using it for queries (and not for queryClient.invalidateQueries, or other queryClient operations). So there's some subtle breaking changes either way.

Reply:
If we want to support a filter that only matches non-finite queries (related: https://github.com/connectrpc/connect-query-es/issues/328), we have to always include a cardinality property with useQuery and useInfiniteQuery, because omitting a property in a query filter means matching any query.

The current approach seems to be a bit confusing, so we should either:
1. Always include cardinality explicitly, which means users have to remember to pass it for exact query key matching.
2. Revert, and only include cardinality for infinite queries, which means non-infinite queries can not be isolated in filters.

Expanding on 1: Currently, all parameters to `createConnectQueryKey` are optional, which makes it very easy to forget passing `cardinality` (or `transport`, for that matter), which won't match with exact query filters. So it could be an option to provide two versions of the function - one that requires all parameters to create a full key, and one that doesn't ("createPartialQueryKey"?), and is can be used for query filters.

Reply:
I prefer 1 for sure, i’m tempted to keep it simple though, without a new API. It feels like just another API that’s so close to the original. An opinionated QueryClient will make the distinction and use cases much more clear and concise.